File: baekjoon/python/브루트 포스/2839_설탕 배달.py
n = int(input())

ans = -1
for i in range(0, n//5+1):
    for j in range(0, n//3+1):
        if((5*i + 3*j) == n):
            ans = i+j if ans == -1 else min(ans, i+j)
print(ans)

File: baekjoon/js/Dynamic Programming/2839_설탕 배달.js
const filePath = process.platform === "linux" ? "/dev/stdin" : "../input.txt";
const input = require("fs").readFileSync(filePath).toString().trim();

const n = +input;
const arr = Array(5001).fill(1700);
arr[3] = 1;
arr[5] = 1;

for(let x = 6; x <= n; x++){
    arr[x] = Math.min(arr[x-3]+1, arr[x-5]+1);
}
console.log(arr[n] >= 1700 ? -1 : arr[n]);
